pkg/sources/github/github.go

```diff
@@ -128,6 +130,59 @@ func (s *Source) UserAndToken(ctx context.Context, installationClient *github.Cl
 	return "", "", errors.New("unhandled credential type for token fetch")
 }
 
+// filteredRepoCache is a wrapper around cache.Cache that filters out repos
+// based on include and exclude globs.
+type filteredRepoCache struct {
+	cache.Cache
+	include, exclude []string
+}
+
+func newFilteredRepoCache(c cache.Cache, include, exclude []string) *filteredRepoCache {
+	return &filteredRepoCache{Cache: c, include: include, exclude: exclude}
+}
+
+// Set overrides the cache.Cache Set method to filter out repos based on
+// include and exclude globs.
+func (c *filteredRepoCache) Set(key, val string) {
+	if c.ignoreRepo(key) {
+		return
+	}
+	if !c.includeRepo(key) {
+		return
+	}
+	c.Cache.Set(key, val)
+}
+
+func (c *filteredRepoCache) ignoreRepo(s string) bool {
+	for _, exclude := range c.exclude {
+		g, err := glob.Compile(exclude)
+		if err != nil {
+			continue
```

Comment:
the old code logged glob pattern compilation errors, but this new version silently drops them. are they being logged elsewhere somehow?

Reply:
They should be logged now during init.